### ISSUE TITLE
grid and downscaling images then predictions at scale

### ISSUE THREAD

**Issue by Nal44** [mpicbg-csbd/stardist#118]
Hi,

I am training startdist 3D with images taken at 63x . 
The sizes are (50, 1440, 1920) it was takin too much memory and the FOV was too small so I downscale to (50, 240, 320) .
It is learning ok , but the median size object is [ 37.5 156.  160.5] and the FOV (MAX before OOM)  is [ 64 128 128].

I increased the grid and the unet to 3 .
After learning, when I apply to predictions on new images the issues is that the Z is right (because below 64) but the X and Y are way too small because of the FOV of [ 64 128 128] .

Hence it can only see object below that shape : so many small nucleus detected on 1 single nucleus 
Hence my question is how to use Stardist when trained on downscale object onto bigger images with full scale (50, 1440, 1920) ?
 
Could it be possible to downscale, detect and rescale for the predictions ?
Is that clear or not ? (not sure if I explained well! ).

Thanks a lot,

**Comment by maweigert** [comment]
Hi @Nal44 ,

I am not entirely sure if I understand everything correctly, but if you downscale the training data by a certain factor, you should do the same for at the prediction step, i.e. you would have to do

> downscale, detect and rescale for the predictions ?

by yourself. 

Hope that helps,

M

**Comment by uschmidt83** [comment]
> the median size object is [ 37.5 156. 160.5]

Before or after downscaling the input images?

> I increased the grid and the unet to 3 .
> After learning, when I apply to predictions on new images the issues is that the Z is right (because below 64) but the X and Y are way too small because of the FOV of [ 64 128 128] .
> 
> Hence it can only see object below that shape : so many small nucleus detected on 1 single nucleus
> Hence my question is how to use Stardist when trained on downscale object onto bigger images with full scale (50, 1440, 1920) ?

It is difficult to understand what you mean. Ideally show us the code/notebook.

**Comment by Nal44** [comment]
Hi Both,
Before or after downscaling the input images : After downscaling.
It is difficult to understand what you mean. Ideally show us the code/notebook : sure ! 

I saved it as .html and zip it , if that helps, if not let me know how to share it .

I added the predictions too.

Thanks a lot,

[prediction tiles-Copy1 cells 63x 3d test.zip](https://github.com/mpicbg-csbd/stardist/files/6043784/prediction.tiles-Copy1.cells.63x.3d.test.zip)

antho
[prediction tiles-Copy1 cells 63x 3d test.zip](https://github.com/mpicbg-csbd/stardist/files/6043787/prediction.tiles-Copy1.cells.63x.3d.test.zip)
[CELL VARIOUS 63x 3d-Copy1 with synthetic-bigger grid trial1.zip](https://github.com/mpicbg-csbd/stardist/files/6043788/CELL.VARIOUS.63x.3d-Copy1.with.synthetic-bigger.grid.trial1.zip)



 
 
[CELL VARIOUS 63x 3d-Copy1 with synthetic-bigger grid trial1.zip](https://github.com/mpicbg-csbd/stardist/files/6043769/CELL.VARIOUS.63x.3d-Copy1.with.synthetic-bigger.grid.trial1.zip)

**Comment by Nal44** [comment]
In the CELL VARIOUS 63x 3d-Copy1 with synthetic-bigger grid trial1 I did the training with unet 3 and the grid at 2,4,4  but then I tried to increase the FOV so I changed to unet 4 and grid, hence the OOM error that would will see.
But it trained well with unet 3 and grid 2,4,4. but the objects detection are too small when using the predictions on new images .

Thanks for the support and help !
antho

**Comment by maweigert** [comment]
What you see at prediction time is expected, as your training didn't result in a well performing model: If you have a look into your training notebook, you will see that you final accuracy is ~0.1 which is far from anything useful. 
Try to 
1) downsample tou data along y and x (e.g. 4 fold) before training, and 
2) using `backbone=unet`.

**Comment by Nal44** [comment]
I did downscale 6 fold on X and Y before training and use the unet backbone ... ?
Should I keep the depth to 2 ? what about the grid ? keep the same ?

The issue is that the median object will be larger and the FOV will be smaller I will get the  #WARNING: median object size larger than field of view of the neural network.# .

I am puzzled and cant find the right parameters, perhaps could I send you my training files to check if you can find the right parameters , I may have done something wrong with the annotation because the median object is very large after downscaling isn't it ?
Let me know and thanks for you help !
